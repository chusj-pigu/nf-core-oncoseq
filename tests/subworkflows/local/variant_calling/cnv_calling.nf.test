nextflow_workflow {

    name "Test Workflow CNV_CALLING"
    script "subworkflows/local/variant_calling/cnv_calling.nf"
    workflow "CNV_CALLING"

    test("Should run without failures") {

        when {
            params {
                // define parameters here. Example:
                // outdir = "tests/results"
            }
            workflow {
                """
                // define inputs of the workflow here. Example:
                ch_bam = Channel.fromPath("${projectDir}/test_data/variant_calling/cnv_calling/sample1.sorted.bam")
                ch_bai = Channel.fromPath("${projectDir}/test_data/variant_calling/cnv_calling/sample1.sorted.bam.bai")
                ch_input0 = ch_bam
                    .combine(ch_bai)
                    .map { bam, bai ->
                        tuple(id:'sample1',bam,bai)}
                ch_ref = Channel.fromPath("${projectDir}/test_data/realistic_data/hg38.fa*")
                    .toSortedList()
                    .map { ref, ref_fai ->
                        tuple(id:"sample1", "hg38", ref, ref_fai) }
                input[0] = ch_input0
                input[1] = ch_ref
                """
            }
        }

        then {
            assert workflow.success
        }

    }

}
