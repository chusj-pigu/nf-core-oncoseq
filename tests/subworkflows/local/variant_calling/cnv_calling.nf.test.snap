{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "sample1"
                        },
                        "sample1_cnv_calls.vcf:md5,b0224052d5268b140cf5fc33445c52e2"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "sample1"
                        },
                        "sample1_cnv_cov.png:md5,848d09fe7ca4cf7b668b87402ef48cd0"
                    ]
                ],
                "2": [
                    "versions.yml:md5,e1b620736ba90ba0fefc07ad9ea456f5"
                ],
                "qdnaseq_plot": [
                    [
                        {
                            "id": "sample1"
                        },
                        "sample1_cnv_cov.png:md5,848d09fe7ca4cf7b668b87402ef48cd0"
                    ]
                ],
                "qdnaseq_vcf": [
                    [
                        {
                            "id": "sample1"
                        },
                        "sample1_cnv_calls.vcf:md5,b0224052d5268b140cf5fc33445c52e2"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,e1b620736ba90ba0fefc07ad9ea456f5"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.2"
        },
        "timestamp": "2025-04-11T21:22:59.723021585"
    }
}